bobtimus/src/bin/bobtimus.rs

```diff
@@ -0,0 +1,204 @@
+use anyhow::Result;
+use bobtimus::{cli::StartCommand, routes, Bobtimus};
+use elements_fun::secp256k1::rand::{rngs::StdRng, thread_rng, SeedableRng};
+use elements_harness::Client;
+use structopt::StructOpt;
+use warp::Filter;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    env_logger::init();
+
+    let StartCommand {
+        elementsd_url,
+        api_port,
+        usdt_asset_id,
+    } = StartCommand::from_args();
+
+    let elementsd = Client::new(elementsd_url.into_string())?;
+    let btc_asset_id = elementsd.get_bitcoin_asset_id().await?;
+
+    let rate_service = kraken::RateService::new().await?;
+
+    let bobtimus = Bobtimus {
+        rng: StdRng::from_rng(&mut thread_rng()).unwrap(),
+        rate_service,
+        elementsd,
+        btc_asset_id,
+        usdt_asset_id,
+    };
+
+    let bobtimus_filter = warp::any().map({
+        let bobtimus = bobtimus.clone();
+        move || bobtimus.clone()
+    });
+
+    let subscription = bobtimus.rate_service.subscribe();
+    let latest_rate = warp::path!("rate" / "lbtc-lusdt")
+        .and(warp::get())
+        .map(move || routes::latest_rate(subscription.clone()));
+
+    let create_swap = warp::post()
+        .and(warp::path!("swap" / "lbtc-lusdt"))
+        .and(warp::path::end())
+        .and(bobtimus_filter.clone())
+        .and(warp::body::json())
+        .and_then(routes::create_swap);
+
+    warp::serve(latest_rate.or(create_swap))
+        .run(([127, 0, 0, 1], api_port))
+        .await;
+
+    Ok(())
+}
+
+mod kraken {
+    use anyhow::{anyhow, bail, Result};
+    use async_trait::async_trait;
+    use bobtimus::{LatestRate, LiquidUsdt, Rate};
+    use futures::{SinkExt, Stream, StreamExt};
+    use reqwest::Url;
+    use serde::{Deserialize, Serialize};
+    use serde_json::Value;
+    use std::convert::TryFrom;
+    use tokio::sync::watch;
+    use tokio_tungstenite::tungstenite::Message;
+    use watch::Receiver;
+
+    const KRAKEN_WS_URL: &str = "wss://ws.kraken.com";
+    const SUBSCRIBE_XBT_USDT_TICKER_PAYLOAD: &str = r#"
+    { "event": "subscribe",
+      "pair": [ "XBT/USD" ],
+      "subscription": {
+        "name": "ticker"
+      }
+    }"#;
```

Comment:
Nitpick: Either rename variable to `...XBT_USD_TICKER` or subscribe to `XBT/USDT` because  
`USD` is the fiat asset while `USDT` is the crypto asset (the stable coin).

```suggestion
    const SUBSCRIBE_XBT_USDT_TICKER_PAYLOAD: &str = r#"
    { "event": "subscribe",
      "pair": [ "XBT/USDT" ],
      "subscription": {
        "name": "ticker"
      }
    }"#;
```

Reply:
Good point! I think I'll go for the rename, because I tried the pair `XBT/USDT` and the subscription produced very few updates over time.